source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2515 @@
+===================
+Unified Test Format
+===================
+
+:Spec Title: Unified Test Format
+:Spec Version: 1.0.0
+:Author: Jeremy Mikola
+:Advisors: Prashant Mital, Isabel Atkinson, Thomas Reggi
+:Status: Draft
+:Type: Standards
+:Minimum Server Version: N/A
+:Last Modified: 2020-09-01
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+This project defines a unified schema for YAML and JSON specification tests,
+which run operations against a MongoDB deployment. By conforming various spec
+tests to a single schema, drivers can implement a single test runner to execute
+acceptance tests for multiple specifications, thereby reducing maintenance of
+existing specs and implementation time for new specifications.
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`__.
+
+Specification
+=============
+
+
+Terms
+-----
+
+Entity
+  Any object or value that is indexed by a unique name and stored in the
+  `Entity Map`_. This will typically be a driver object (e.g. client, session)
```

Comment:
I have a concern related to the entity map and scope. Currently `createEntities` creates these top-level entities with seemingly global scope. `saveResultAsEntity` within a `test.operations` `operation` can define and add entities to the global entities, how do we handle overwriting entities? How do we handle the scoping of the test-specific entities, and prevent leaking?